src/cpp/session/SessionMain.cpp

```diff
@@ -1878,10 +1878,10 @@ int main (int argc, char * const argv[])
       FilePath userScratchPath = options.userScratchPath();
       if (userScratchPath.exists())
       {
-         std::vector<FilePath> scratchChildren;
-         userScratchPath.getChildren(scratchChildren);
-
-         if (scratchChildren.size() == 0)
+         // if the lists directory has not yet been created,
+         // this is a new user
+         FilePath listsPath = userScratchPath.completeChildPath("monitored/lists");
```

Comment:
nit: may want to use `monitoredParentPath()` here; although it would require making this function public:

https://github.com/rstudio/rstudio/blob/c65a73b21833d0e5f9a3e25ef85b038dd36d5113/src/cpp/session/SessionModuleContext.cpp#L543-L550